### ISSUE TITLE
Flutter in Debug JIT mode OOMs on Fuchsia

### ISSUE THREAD

**Issue by apwilson** [flutter/flutter#59113]
Found while working on https://b.corp.google.com/issues/158690361, this issue corresponds to  https://b.corp.google.com/issues/158312525.

**Comment by kf6gpe** [comment]
Currently still under investigation by @iskakaushik and @rmacnak-google. Latest comment on upstream bug is OK to cross-post here, and is from 15-Jun:
```
In that snapshot, there's ~85MB of Strings that are retained by Scripts. That suggests the whole source for the program is taking up about half of the heap. It is not needed for execution, but some debugger tooling may want it.
```

**Comment by kf6gpe** [comment]
As an update, @iskakaushik  been unable to reproduce this reliably, and was blocked on some issues around the target's build for much of the time since then. He'll be unavailable for the rest of the week, so there won't be forward progress on this during that time.

**Comment by kf6gpe** [comment]
@iskakaushik has a reproducible case and is actively working on this issue.

**Comment by chinmaygarde** [comment]
@iskakaushik If there anything to do w.r.t this issue other than investigating how to trim the source string from the program? Other than that, it seems WAI that debug mode application are more resource intensive than release-mode ones.

**Comment by Hixie** [comment]
We need to make sure that our colleagues can debug their app. How exactly we help them do that (e.g. splitting the app into smaller apps, or making Flutter more memory efficient, or something else) is up to them and us to figure out.

**Comment by iskakaushik** [comment]
Update: David Murphy @ Google is attempting to change the build rules to not embed the sources and see if that lets the customer run the app in jit-debug mode. I will keep this assigned to myself so if the proposed fix doesn't work, we might need to explore other ways.

**Comment by iskakaushik** [comment]
The proposed fix of `--no-embed-sources` has worked. Closing this issue.
